zebra-state/src/service/finalized_state/disk_format/chain.rs

```diff
@@ -93,3 +94,26 @@ impl FromDisk for HistoryTreeParts {
             .expect("deserialization format should match the serialization format used by IntoDisk")
     }
 }
+
+impl IntoDisk for BlockData {
+    type Bytes = Vec<u8>;
+
+    fn as_bytes(&self) -> Self::Bytes {
+        self.value_pools().as_bytes().to_vec()
```

Comment:
Nitpick: It'd be nice to use the exact size if we know it in advance and move it to constants.

```suggestion
    type Bytes = [u8; 40];

    fn as_bytes(&self) -> Self::Bytes {
        self.value_pools().as_bytes()
```

Reply:
If we use `Vec` then this code will be forward-compatible - we can append more data without requiring a full database upgrade. Not sure if we _want_ that though :sweat_smile: